docs/spec/concepts.rst

```diff
@@ -3,6 +3,268 @@
+or a class that inherits directly or indirectly from ``str``. A :ref:`Protocol
+<Protocols>` denotes the set of all objects which share a certain set of
+attributes and/or methods.
+
+Gradual types
+~~~~~~~~~~~~~
+
+:ref:`Any` represents an unknown static type. It denotes some unknown set of
+values.
+
+This may appear similar to the fully static type ``object``, which represents
+the set of all Python objects, but it is quite different.
+
+If an expression has the type ``object``, a static type checker should ensure
+that operations on the expression are valid for all Python objects, or else
+emit a static type error. This allows very few operations! For example, if
+``x`` is typed as ``object``, ``x.foo`` should be a static type error because
+not all Python objects have an attribute ``foo``.
+
+An expression typed as :ref:`Any`, on the other hand, should be assumed to have
+_some_ specific static type, but _which_ static type is not known. A static
+type checker should not emit static type errors on an expression or statement
+if :ref:`Any` might represent a static type which would avoid the error. (This
+intuition is made more precise below, in our definitions of materialization and
+assignability.)
+
+Similarly, a gradual type such as ``tuple[int, Any]`` (see :ref:`tuples`) or
+``int | Any`` (see :ref:`union-types`) does not represent a single set of
+Python objects; rather, it represents a (bounded) range of possible sets of
+values.
+
+In the same way that ``Any`` does not represent "the set of all Python objects"
+but rather "an unknown set of objects", ``tuple[int, Any]`` does not represent
+"the set of all length-two tuples whose first element is an integer." That is a
+fully static type, spelled ``tuple[int, object]``.  In contrast, ``tuple[int,
+Any]`` represents some unknown set of tuple values; it might be the set of all
+tuples of two integers, or the set of all tuples of an integer and a string, or
+some other set of tuple values.
+
+In practice, this difference is seen (for example) in the fact that we can
+assign an expression of type ``tuple[int, Any]`` to a target typed as
+``tuple[int, int]``, whereas assigning ``tuple[int, object]`` to ``tuple[int,
+int]`` is a static type error. (Again, we formalize this distinction in the
+below definitions of materialization and assignability.)
+
+In the same way that the fully static type ``object`` is the upper bound for
+the possible sets of values represented by ``Any``, the fully static type
+``tuple[int, object]`` is the upper bound for the possible sets of values
+represented by ``tuple[int, Any]``.
+
+The gradual guarantee
+~~~~~~~~~~~~~~~~~~~~~
+
+:ref:`Any` allows gradually adding static types to a dynamically-typed program.
+For a fully dynamically-typed program, a static checker has the type :ref:`Any`
+for all expressions, and should emit no errors. Adding static type annotations
+to the program (making the program more statically typed) may result in static
```

Comment:
I think it is useful, because it calls back to our earlier definitions of "statically typed" and "dynamically typed."

Reply:
It might read a bit better if you rephrased it to "Adding type annotations" rather than "adding _static_ type annotations". That would mean you wouldn't have "static" twice in the same sentence (and I don't think there's any other kind of type annotation in Python, other than static type annotations)